src/ui/citation-popup/index.ts

```diff
@@ -46,17 +51,16 @@ async function loadAssets (refs: Reference[]): Promise<void> {
     await joplin.views.dialogs.addScript(popupHandle, "./ui/citation-popup/view.js");
 }
 
-function fromRefsToHTML (refs: Reference[]): string {
-    const JSONString = JSON.stringify(
-        refs.map(ref => {
-            return {
-                id: ref.id,
-                title: ref.title,
-                author: ref.author,
-                year: (ref.issued && ref.issued["date-parts"]) ? getDate(ref).getFullYear() : null
-            };
-        })
-    );
+function toHTML (req: Request): string {
+    const refs = req.refs.map(ref => {
+        return {
+            id: ref.id,
+            title: ref.title,
+            author: ref.author,
+            year: (ref.issued && ref.issued["date-parts"]) ? getDate(ref).getFullYear() : null
```

Comment:
That's a leaky abstraction here - you shouldn't access the "date-parts" key anywhere since your helper function is supposed to deal with this. Either do `const fullDate = getDate(ref);`, then later `year: fullDate ? fullDate.getFullYear() : null` or create another helper function like `getDateYear(ref)` that does the date-parts check and either return the year or null.

Reply:
Noted